server/datastore/mysql/software.go

```diff
@@ -509,18 +509,35 @@ func (ds *Datastore) AllCPEs(ctx context.Context) ([]string, error) {
 // provided cpes. It returns the number of new rows inserted or an error. If
 // the CVE already existed for all CPEs, it would return 0, nil.
 func (ds *Datastore) InsertCVEForCPE(ctx context.Context, cve string, cpes []string) (int64, error) {
-	values := strings.TrimSuffix(strings.Repeat("((SELECT id FROM software_cpe WHERE cpe=?),?),", len(cpes)), ",")
-	sql := fmt.Sprintf(`INSERT IGNORE INTO software_cve (cpe_id, cve) VALUES %s`, values)
-	var args []interface{}
+	var totalCount int64
 	for _, cpe := range cpes {
-		args = append(args, cpe, cve)
-	}
-	res, err := ds.writer.ExecContext(ctx, sql, args...)
-	if err != nil {
-		return 0, ctxerr.Wrap(ctx, err, "insert software cve")
+		var ids []uint
+		query, args, err := sqlx.In(`SELECT id FROM software_cpe WHERE cpe=?`, cpe)
```

Comment:
Why are you using `sqlx.In` here? You are not passing any args that are slices.

Reply:
oh, copy paste error gah